pkg/plugin/manager/manager.go

```diff
@@ -149,11 +193,20 @@ func (m *manager) fixAndRun(path string) error {
 			return errors.New(fmt.Sprintf("could not open db: %s", setErr.Error()))
 		}
 	}
+
+	_, execErr := m.db.Exec(fmt.Sprintf("CREATE SCHEMA IF NOT EXISTS %s", m.GenConfig.Schema))
+	if execErr != nil {
+		return execErr
+	}
+
+	goose.SetTableName(fmt.Sprintf("%s.goose_db_version", m.GenConfig.Schema))
```

Comment:
More thinking out loud here than anything else - but it looks like this means we're assuming that we're only dealing with one plugin anytime this is invoked. That seems valid, if we're saying each plugin is responsible for running its own `execute` process, but just want to make sure we're all locked in on/documenting that understanding - don't compose multiple plugins, create a separate schema/process for each.

lmk if that's wrong and if this could be compatible with multiple plugin schemas at once

Reply:
If indeed we're assuming one plugin, then I wonder if we can cut out any complexity here. I think this package was built with the idea that we'd be assembling multiple plugins (which is why the config has a rank for migrations, etc)

Reply:
I was assuming one plugin because I didn't realize there could be more. I'm honestly not sure how multiple would work. We'd need to migrations to separate directories and fixAndRun those separately, separated by schema. 

Okay I lied I guess I do know how it would work. I'm not sure it's worthwhile - to a point you've made before should we really be running migrations here when we deploy with docker anyway?

Reply:
I agree we shouldn't spend too much time thinking about running migrations here since it's not how it's used in the deploy, but was mostly just calling out that I think this is - in the abstract - structured to manage and run migrations from multiple plugins right now. So if we're deprecating that, maybe there's an opportunity to delete some code/remove some complexity

Reply:
I'm kinda asking if we should.

Upon further reflection, I think this would work with multiple plugins, provided they shared the same schema name.  Maybe document that?

Reply:
yeah but why would multiple plugins share a schema? I feel like the one clear thing that separates different plugins is the schemas they manage